falcon/media/json.py

```diff
@@ -2,30 +2,79 @@
 
 from falcon import errors
 from falcon.media import BaseHandler
-from falcon.util import compat
 from falcon.util import json
 
+from functools import partial
+
 
 class JSONHandler(BaseHandler):
     """JSON media handler.
 
     This handler uses Python's :py:mod:`json` by default, but will
     use :py:mod:`ujson` if available.
+
+    Keyword Arguments:
+        dumps (func): the function to use when serializing JSON responses.
+
+        loads (func): the function to use when deserializing JSON requests.
+
+    You can override the JSON library used by changing the ``dumps`` and
+    ``loads`` functions. Some good options are orjson, python-rapidjson,
+    and mujson.::
+
+        import falcon
+        from falcon import media
+
+        import rapidjson
+
+        json_handler = media.JSONHandler(
+            dumps=rapidjson.dumps,
+            loads=rapidjson.loads,
+        )
+        extra_handlers = {
+            'application/json': json_handler,
+        }
+
+        api = falcon.API()
+        api.req_options.media_handlers.update(extra_handlers)
+        api.resp_options.media_handlers.update(extra_handlers)
+
+
+    By default, ``ensure_ascii`` is passed to the ``json.dumps`` function.
+    If you override the dumps function you might want to include that as a
+    default parameter. A simple way is by using ``functools.partial`` to curry
```

Comment:
I think I might have misused the term *to curry* usage in my earlier comment, sorry about that: https://en.wikipedia.org/wiki/Currying#Contrast_with_partial_function_application